common/common-client/src/main/java/org/ow2/proactive/authentication/IAMLoginModule.java

```diff
@@ -0,0 +1,392 @@
+                logger.debug("Set jsonWebTokenSigned=" + jsonWebTokenSigned);
+            } else if ("jsonWebTokenEncrypted".equals(entry.getKey())) {
+                this.jsonWebTokenEncrypted = Boolean.parseBoolean((String) entry.getValue());
+                logger.debug("Set jsonWebTokenEncrypted=" + jsonWebTokenEncrypted);
+            } else if ("jsonWebTokenSignatureKey".equals(entry.getKey())) {
+                this.jsonWebTokenSignatureKey = (String) entry.getValue();
+                logger.debug("Set jsonWebTokenSignatureKey=" + jsonWebTokenSignatureKey);
+            } else if ("jsonWebTokenEncryptionKey".equals(entry.getKey())) {
+                this.jsonWebTokenEncryptionKey = (String) entry.getValue();
+                logger.debug("Set jsonWebTokenEncryptionKey=" + jsonWebTokenEncryptionKey);
+            }
+
+        }
+
+        CommonUtils.assertNotNull(ticketValidatorClass, "ticketValidatorClass is required.");
+        this.ticketValidator = createTicketValidator(ticketValidatorClass, options);
+    }
+
+    /**
+     * Authenticate the user by getting the user name and password from the
+     * CallbackHandler.
+     *
+     * <p>
+     *
+     * @return true in all cases since this <code>IAMLoginModule</code>
+     *         should not be ignored.
+     *
+     * @exception FailedLoginException
+     *                if the authentication fails.
+     *                <p>
+     *
+     * @exception LoginException
+     *                if this <code>IAMLoginModule</code> is unable to
+     *                perform the authentication.
+     */
+    @Override
+    public boolean login() throws LoginException {
+
+        succeeded = false;
+
+        if (callbackHandler == null) {
+            throw new LoginException("Error: no CallbackHandler available " +
+                                     "to garner authentication information from the user");
+        }
+        try {
+            Callback[] callbacks = new Callback[] { new NoCallback() };
+
+            // gets the username, password, group Membership, and group Hierarchy from callback handler
+            callbackHandler.handle(callbacks);
+            Map<String, Object> params = ((NoCallback) callbacks[0]).get();
+            String username = (String) params.get("username");
+            String password = (String) params.get("pw");
+
+            params.clear();
+            ((NoCallback) callbacks[0]).clear();
+
+            CommonUtils.assertNotNull(username, "No username has been specified for authentication");
+            CommonUtils.assertNotNull(password, "No password has been specified for authentication");
+
+            assertion = ticketValidator.validate(getServiceToken(username, password), service);
+
+            CommonUtils.assertNotNull(assertion,
+                                      "Assertion returned by ticket validator " + this.ticketValidatorClass +
+                                                 " is null");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal " is null" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)